src/main/java/org/benetech/servicenet/service/impl/OrganizationServiceImpl.java

```diff
@@ -1112,7 +1112,24 @@ private void saveDatesClosed(Map<Integer, List<String>> datesClosedByLocation, L
                 eligibilityService.save(clonedEligibility);
                 srvClone.setEligibility(clonedEligibility);
             }
-            clonedServices.add(serviceService.save(srvClone));
+
+            Set<ServiceAtLocation> sats = new HashSet<>();
+            for (ServiceAtLocation sat : service.getLocations()) {
+                Optional<Location> loc = locations.stream()
+                    .filter(l -> l.getExternalDbId() != null
+                        && l.getExternalDbId().startsWith(sat.getLocation().getExternalDbId()))
+                    .findFirst();
+                if (loc.isPresent()) {
+                    ServiceAtLocation serviceAtLocation = new ServiceAtLocation();
+                    serviceAtLocation.setProviderName(SERVICE_PROVIDER);
+                    serviceAtLocation.setLocation(loc.get());
```

Comment:
I think the cloned location should be set here instead of the original service's location.

Reply:
the cloned locations are passed as a 'locations' param